services/federator/src/Federator/Service.hs

```diff
@@ -38,9 +38,11 @@ import Wire.API.Federation.Domain (originDomainHeaderName)
 newtype ServiceError = ServiceErrorInvalidStatus HTTP.Status
   deriving (Eq, Show)
 
-data Service m a where
+type ServiceLBS = Service (Maybe LByteString)
+
+data Service body m a where
   -- | Returns status and body, 'HTTP.Response' is not nice to work with in tests
-  ServiceCall :: Component -> ByteString -> LByteString -> Domain -> Service m (HTTP.Status, Maybe LByteString)
+  ServiceCall :: Component -> ByteString -> LByteString -> Domain -> Service body m (HTTP.Status, body)
```

Comment:
Why is the additional body argument needed for the `Service` type constructor? In the definition of `ServiceCall`, couldn't you have said it is of type `Service m (HTTP.Status, a)`?

Reply:
Because otherwise any interpreter should be able to return an `a` as a body for all possible types `a`, which clearly won't work. The type parameter here is not that important, since the effect is currently only used with `a = StreamingResponse`.